cpp/include/cudf/functions.h

```diff
@@ -894,6 +894,60 @@ gdf_error gdf_group_by_count(int ncols,                    // # columns
                              gdf_column* out_col_agg,      //aggregation result
                              gdf_context* ctxt);            //struct with additional info: bool is_sorted, flag_sort_or_hash, bool flag_count_distinct
 
+/* --------------------------------------------------------------------------*/
+  /**
+   * @brief Given a set of columns, of which a subset of these are defined to be the group by columns,
+   * all columns are sorted by the group by columns and returned, along with a column containing
+   * a list of the start indices of each group
+   *
+   * @Param[in] The number of columns in the dataset
+   * @Param[in] The input columns in the dataset
+   * @Param[in] The number of columns to be grouping by
+   * @Param[in] The column indices of the input dataset that will be grouped by
+   * @Param[out] The dataset sorted by the group by columns (needs to be pre-allocated)
+   * @Param[out] A column containing the starting indices of each group. Indices based off of new sort order. (needs to be pre-allocated)
```

Comment:
Somewhat confusing:
* We already have at least two kinds of indices: An index into a data and an index into the sequence of columns. Is this a new kind?
* "Indices based off of new sort order" - This is not a sentence. Even if it were to say "indices are based off of the new sort order" - this is still not clear. Offsets are "based off of" things. "New sort order" - it's not clear to me what order that is exactly.
* "Needs to be pre-allocated" - How much space is necessary exactly?